### ISSUE TITLE
Type error exception when handling meter groups when metadata for meter contains a room

### ISSUE THREAD

**Issue by gjwo** [nilmtk/nilmtk#409]
See https://github.com/nilmtk/nilmtk/issues/405 for context
```python
from nilmtk import DataSet
gjw = DataSet('/Users/GJWood/nilm_gjw_data/HDF5/nilm_gjw_data.hdf5')
elec = gjw.buildings[1].elec
elec

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
c:\Users\GJWood\Anaconda\lib\site-packages\IPython\core\formatters.pyc in __call__(self, obj)
    693                 type_pprinters=self.type_printers,
    694                 deferred_pprinters=self.deferred_printers)
--> 695             printer.pretty(obj)
    696             printer.flush()
    697             return stream.getvalue()

c:\Users\GJWood\Anaconda\lib\site-packages\IPython\lib\pretty.pyc in pretty(self, obj)
    384                 if cls in self.type_pprinters:
    385                     # printer registered in self.type_pprinters
--> 386                     return self.type_pprinters[cls](obj, self, cycle)
    387                 else:
    388                     # deferred printer

c:\Users\GJWood\Anaconda\lib\site-packages\IPython\lib\pretty.pyc in inner(obj, p, cycle)
    568                 p.text(',')
    569                 p.breakable()
--> 570             p.pretty(x)
    571         if len(obj) == 1 and type(obj) is tuple:
    572             # Special case for 1-item tuples.

c:\Users\GJWood\Anaconda\lib\site-packages\IPython\lib\pretty.pyc in pretty(self, obj)
    399                             if callable(meth):
    400                                 return meth(obj, self, cycle)
--> 401             return _default_pprint(obj, self, cycle)
    402         finally:
    403             self.end_group()

c:\Users\GJWood\Anaconda\lib\site-packages\IPython\lib\pretty.pyc in _default_pprint(obj, p, cycle)
    519     if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
    520         # A user-provided repr. Find newlines and replace them with p.break_()
--> 521         _repr_pprint(obj, p, cycle)
    522         return
    523     p.begin_group(1, '<')

c:\Users\GJWood\Anaconda\lib\site-packages\IPython\lib\pretty.pyc in _repr_pprint(obj, p, cycle)
    701     """A pprint that just redirects to the normal repr function."""
    702     # Find newlines and replace them with p.break_()
--> 703     output = repr(obj)
    704     for idx,output_line in enumerate(output.splitlines()):
    705         if idx:

c:\users\gjwood\nilmtk\nilmtk\elecmeter.pyc in __repr__(self)
    307         room = self.metadata.get('room')
    308         if room:
--> 309             string += ', room={}'.format(room['name'])
    310             try:
    311                 string += '{:d}'.format(room['instance'])

TypeError: string indices must be integers, not str
```
This is the relevant metadata, with the room line uncommented the error occurs
```yaml

rooms:
- {floor: 0, instance: 1, name: lounge}
- {floor: 0, instance: 1, name: hall}
- {floor: 0, instance: 1, name: kitchen}
- {floor: 0, instance: 1, name: utility}
...
elec_meters:
  1:
    site_meter: true
    name: house_total
    #room: utility
    submeter_of: 0
    device_model: ED01013  # keys into meter_devices dictionary
    data_location: /building1/elec/meter1
```

**Comment by JackKelly** [comment]
fixed in 0d6cdb9

(you should now be able to uncomment `#room: utility` :) )

**Comment by gjwo** [comment]
I reinstated the room and can confirm it is fixed
